velox/functions/lib/window/NthValue.cpp

```diff
@@ -164,6 +229,30 @@ class NthValueFunction : public exec::WindowFunction {
     rowNumbers_[i] = rowNumber <= frameEnd ? rowNumber : kNullRow;
   }
 
+  inline void setRowNumberIgnoreNulls(
+      vector_size_t i,
+      const uint64_t* rawNulls,
+      vector_size_t firstFrame,
+      const vector_size_t* frameStarts,
+      const vector_size_t* frameEnds,
+      vector_size_t offset) {
+    auto frameStart = frameStarts[i];
+    auto frameEnd = frameEnds[i];
+    vector_size_t nonNullCount = 0;
+    for (auto j = frameStart; j <= frameEnd; j++) {
+      if (!bits::isBitSet(rawNulls, j - firstFrame)) {
```

Comment:
firstFrame is the smallest frame in frameStarts. So (j - firstFrame) will not be negative. 

I'll update the name to leastFrame so that there isn't a confusion here on in the other place.